enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -230,6 +231,10 @@ public void obligation( long txId ) throws IOException
             {
                 obligationFulfiller.fulfill( txId );
             }
+            catch ( IllegalStateException e )
+            {
+                throw new ComException( "Failed to pull updates", e );
+            }
```

Comment:
There is plenty of room for redesign here, the question is whether we want to take that hit now or later. With 2.2 out soonish I would say we wait with a bigger overhaul.